test/plugin/ogc/wfs/wfslayerconfig.test.js

```diff
@@ -85,7 +94,31 @@ describe('plugin.ogc.wfs.WFSLayerConfig', function() {
 
     wfs.initializeConfig(config);
     const result = wfs.getBestType(config);
-    expect(result).toBe(0);
+    expect(result).toBe(preferredTypes[0]);
     expect(wfs.params.get('outputformat')).toBe('application/json');
   });
+
+  it('should register new type configs and sort them by priority', function() {
+    plugin.ogc.wfs.WFSLayerConfig.registerType(avroConfig);
+
+    expect(preferredTypes.length).toBe(4);
+    expect(preferredTypes[0]).toBe(avroConfig);
+  });
+
+  it('should select new type configs if they match a layer', function() {
+    const wfs = new plugin.ogc.wfs.WFSLayerConfig();
+    const config = {
+      'url': 'https://example.com/geoserver/ogc',
+      'params': {
+        'typename': 'test#layer1',
+        'outputformat': 'avro/binary'
+      },
+      'formats': ['application/json', 'avro/binary', 'gml3', 'GML2']
+    };
+
+    wfs.initializeConfig(config);
+    const result = wfs.getBestType(config);
+    expect(result).toBe(avroConfig);
+    expect(wfs.params.get('outputformat')).toBe('avro/binary');
+  });
```

Comment:
It might be good to add a test for the `getRequest` change to verify the response type.